### PR TITLE
Allow Digital and SingleAxis input actions for teleport

### PR DESCRIPTION
**Pull request by FejZa**
# XRTK - Mixed Reality Toolkit Pull Request

## Overview

Companion PR to https://github.com/XRTK/XRTK-Core/pull/729

### CODE REVIEW

File: XRTK.SDK/Packages/com.xrtk.sdk/Runtime/Features/UX/Scripts/Pointers/TeleportPointer.cs [hunk 1]
```diff
@@ -212,24 +208,156 @@ public override void OnPostRaycast()
...
+                TeleportValidationResult == TeleportValidationResult.HotSpot)
+                {
+                    canTeleport = true;
+                }
+
+                if (canTeleport)
+                {
+                    teleportEnabled = false;
+
+                    if (TeleportValidationResult == TeleportValidationResult.Valid ||
```
Comment (SimonDarksideJ) [inline XRTK.SDK/Packages/com.xrtk.sdk/Runtime/Features/UX/Scripts/Pointers/TeleportPointer.cs:336]: Could this code section be broken out into its own function as it's repeated quite a few times throughout the script?

**Review by SimonDarksideJ — CHANGES_REQUESTED** [review]

File: XRTK.SDK/Packages/com.xrtk.sdk/Runtime/Features/UX/Scripts/Pointers/TeleportPointer.cs (hunk 1 above)
Comment (FejZa) [reply XRTK.SDK/Packages/com.xrtk.sdk/Runtime/Features/UX/Scripts/Pointers/TeleportPointer.cs:336]: Probably can but wasn't as trivial as I thought and broke dual axis in some cases. I reverted it. Let's keep it like this for now. I have some more work for the TeleportPointer on my list where I will tackle this.

**Comment by SimonDarksideJ** [comment]
First issue.
When the input action for Teleport is changed to something other than Vector two, turning no longer works :S

**Comment by FejZa** [comment]
Not an issue but to be expected. How is it supposed to work with only a single axis / digital input?

File: XRTK.SDK/Packages/com.xrtk.sdk/Runtime/Features/UX/Scripts/Pointers/TeleportPointer.cs [hunk 2]
```diff
@@ -212,24 +208,156 @@ public override void OnPostRaycast()
...
+        {
+            // Only turn off the pointer if we're not the one sending the request
+            if (eventData.Pointer.PointerId == PointerId)
+            {
+                IsTeleportRequestActive = false;
+            }
+            else
+            {
+                IsTeleportRequestActive = true;
+                BaseCursor?.SetVisibility(false);
```
Comment (StephenHodgson) [inline XRTK.SDK/Packages/com.xrtk.sdk/Runtime/Features/UX/Scripts/Pointers/TeleportPointer.cs:292]: Let's do a traditional null check on the unity engine object types

**Comment by SimonDarksideJ** [comment]
Ok, as per discussion, PR is fine as is.  Turning is not supported with Digital or Single Axis.  This will be addressed in 0.3 with a new Locomotion Feature

**Review by SimonDarksideJ — APPROVED** [review]

File: XRTK.SDK/Packages/com.xrtk.sdk/Runtime/Features/UX/Scripts/Pointers/TeleportPointer.cs (hunk 2 above)
Comment (FejZa) [reply XRTK.SDK/Packages/com.xrtk.sdk/Runtime/Features/UX/Scripts/Pointers/TeleportPointer.cs:292]: Agreed. I'll change that in another PR as this code was not introduced in this PR and I don't want to trigger another build on this one. Need this to release to continue testing.

**Pull request merged**